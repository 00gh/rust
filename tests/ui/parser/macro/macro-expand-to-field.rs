#![no_main]

macro_rules! field {
    ($name:ident:$type:ty) => {
        $name:$type
    };
}

macro_rules! variant {
    ($name:ident) => {
        $name
    }
}

struct Struct { //~ NOTE while parsing this struct
    field!(bar:u128), //~ NOTE macros cannot expand to struct fields
    //~^ ERROR expected `:`, found `!`
    //~^^ NOTE expected `:`
    //~^^^ ERROR expected `,`, or `}`, found `(`
}

enum EnumVariant { //~ NOTE while parsing this enum
    variant!(whoops), //~ NOTE macros cannot expand to enum variants
    //~^ ERROR unexpected token: `!`
    //~^^ NOTE unexpected token after this
}
