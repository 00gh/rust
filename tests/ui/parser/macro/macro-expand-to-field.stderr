error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:16:10
   |
LL |     field!(bar:u128),
   |          ^ unexpected token after this
   |
   = note: macros cannot expand to struct fields

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:21:10
   |
LL |     field!(recovers:()),
   |          ^ unexpected token after this
   |
   = note: macros cannot expand to struct fields

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:27:12
   |
LL |     variant!(whoops),
   |            ^ unexpected token after this
   |
   = note: macros cannot expand to enum variants

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:32:12
   |
LL |     variant!(recovers),
   |            ^ unexpected token after this
   |
   = note: macros cannot expand to enum variants

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:39:14
   |
LL |         field!(oopsies:()),
   |              ^ unexpected token after this
   |
   = note: macros cannot expand to struct fields

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:42:14
   |
LL |         field!(oopsies2:()),
   |              ^ unexpected token after this
   |
   = note: macros cannot expand to struct fields

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:50:10
   |
LL |     field!(oopsies:()),
   |          ^ unexpected token after this
   |
   = note: macros cannot expand to union fields

error: unexpected token: `!`
  --> $DIR/macro-expand-to-field.rs:54:10
   |
LL |     field!(recovers:()),
   |          ^ unexpected token after this
   |
   = note: macros cannot expand to union fields

error: aborting due to 8 previous errors

